controller.go

```diff
@@ -321,6 +324,10 @@ func loadSettings() error {
 		return fmt.Errorf("invalid max number of certificates per ALB: %d. AWS does not allow more than %d", maxCertsPerALB, aws.DefaultMaxCertsPerALB)
 	}
 
+	if !strings.Contains("HTTP1 HTTP2 GRPC", targetGroupProtocolVersion) {
+		return fmt.Errorf("invalid target group protocol version: %s. must be one of HTTP1, HTTP2, GRPC", targetGroupProtocolVersion)
+	}
+
```

Comment:
We need to decide also if we would like to support ingress annotation in this scope to enable this individually.
The default value is defined as `"HTTP1"` but before this change we did not set it so I am wondering if changing empty to `"HTTP1"` will cause any issues (e.g. trigger loadbalancer recreation).

Lets check if we can use `kingpin.Enum`. If no it would be still better to check three possible cases instead of this clever trick which e.g. allows `"HTTP1 HTTP2"` or `"TTP"`

Reply:
/cc @szuecs

Reply:
The update behavior of TG is recreating, which is causing a new resource to be created first and the replacing the link. But I'm fine to change accordingly.

Reply:
I'm not sure about annotation support, since this change is invasive and potentially breaking HTTP1.1 clients, why I would see this as a central administrative decision

Reply:
Enum sounds like a good idea to me.
In general it should be the administrator that decides about the default via "flag" and developers can opt-in a different choice via annotation to override the global decision.